core/src/Dscp/Util.hs

```diff
@@ -202,3 +208,9 @@ dieGracefully :: (MonadLogging m, MonadCatch m) => m () -> m ()
 dieGracefully action =
     action `catchAny` \e -> do
         logError $ fromString $ "Exception in transactionRelayInput: " <> show e
+
+-----------------------------------------------------------
+-- Convenient RIO projector
+-----------------------------------------------------------
+
+type Provides x ctx m = (MonadReader ctx m, HasLens' ctx x)
```

Comment:
I doubt its convenience in fact, also the arguments order is weird -- `x` must be the last one imho.